app/src/main/java/app/revanced/tudortmund/lockscreen/ShowOnLockscreenPatch.java

```diff
@@ -0,0 +1,46 @@
+package app.revanced.tudortmund.lockscreen;
+
+import android.content.Context;
+import android.hardware.display.DisplayManager;
+import android.os.Build;
+import android.view.Display;
+import android.view.Window;
+import androidx.appcompat.app.AppCompatActivity;
+
+import static android.view.WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD;
+import static android.view.WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED;
+
+public class ShowOnLockscreenPatch {
+    /**
+     * @noinspection deprecation
+     */
+    public static Window getWindow(AppCompatActivity activity, float brightness) {
+        Window window = activity.getWindow();
+
+        if (brightness >= 0) {
+            // High brightness set, therefore show on lockscreen.
```

Comment:
Please explain what you mean by this.

Reply:
Whenever we open the student id or student ticket in the app, the brightness gets set to the maximum. That's why we check for when the brightness gets set and to what value. If it gets set to increase the brightness, we activate that the activity is shown on the lockscreen. And if the brightness gets reset, we disable it.